src/source/TsGenerator.scala

```diff
@@ -253,7 +253,7 @@ class TsGenerator(spec: Spec) extends Generator(spec) {
         case r: Record => generateRecord(td.origin, td.ident, td.doc, td.params, r, w)
         case i: Interface => {
           generateInterface(td.origin, td.ident, td.doc, td.params, i, w)
-          if (i.methods.exists(_.static)) {
+          if (i.methods.exists(m => m.static && m.lang.js)) {
```

Comment:
Great example of why tests matter. The earlier we catch the issues, the more time we save